zetta_utils/db_annotations/precomp_annotations.py

```diff
@@ -592,14 +722,26 @@ def post_process(self):
         This is useful after writing out a bunch of data with
           write_annotations(data, False), which writes to only the lowest-level chunks.
         """
-        # read data (from lowest level chunks)
-        all_data = self.read_all()
-
-        # write data to all levels EXCEPT the last one
-        levels_to_write = range(0, len(self.chunk_sizes) - 1)
-        spatial_entries = subdivide(
-            all_data, self.index, self.chunk_sizes, self.path, levels_to_write
-        )
+        if len(self.chunk_sizes) == 1:
+            # Special case: only one chunk size, no subdivision.
+            # In this case, we can cheat considerably.
+            # Just iterate over the spatial entry files, getting the line
+            # count in each one, and keep track of the max.
+            max_line_count = self.find_max_size(0)
+            print(f"Found max_line_count = {max_line_count}")
+            spatial_entries = self.get_spatial_entries(max_line_count)
+        else:
+            # Multiple chunk sizes means we have to start at the lowest
+            # level, and re-subdivide it at each higher level.
+
+            # read data (from lowest level chunks)
+            all_data = self.read_all()
```

Comment:
Same as for the size requests above - when loading annotations from object storage, you will get a considerable speedup by downloading all objects in one batch, rather than one by one as `read_all` / `read_lines` currently do.

Reply:
That's a good point.  In this case, though, file I/O is not the bottleneck (even over the network), and refactoring it to download all at once would require a pretty major redesign.